### DISCUSSION TITLE
Trigger event "field-change" for FileField

### DISCUSSION THREAD

**Discussion by lisotton** [laravel/nova-issues#4926] [Ideas & Feature Requests]
Hi, 

Most of the fields trigger an event `Nova.$emit(this.field.attribute + '-change'...` when there is a change to the input, but it is not the case for `FileField`, there are some events when using `VaporFile`, but I think should emit an event for `FileField` as well.

Best regards,

Leandro Isotton

**Comment by davidhemphill** [comment]
What's your use-case for this?

**Comment by lisotton** [comment]
I have a resource for Podcasts and I would like to extract the duration of the audio to populate the field `duration` after the file is selected.

**Comment by crynobone** [comment]
> I have a resource for Podcasts and I would like to extract the duration of the audio to populate the field duration after the file is selected.

So the event has to emit the full file? Other field changes event just send scalar values (mostly string/array etc).

**Comment by lisotton** [comment]
Yes @crynobone, for example in my case, to be able to use `FileReader`, I need the `event.currentTarget.files[0]`.
